### ISSUE TITLE
IL2CPP support

### ISSUE THREAD

**Issue by develmusa** [ViRGIS-Team/gdal-upm#6]
To make GDAL available in Unity for all platforms, IL2CPP is required.

Is it possible to extend the project for IL2CPP?

**Comment by runette** [comment]
Unfortunately - the GDAL C# bindings are made using SWIG and there are some open issues that prevent SWIG working in IL2CPP (for instance https://github.com/swig/swig/issues/1165).

I believe that there are probably work arounds that could be put in place but that would involve changing the GDAL bindings in GDAL - which would be a huge and complicated job and I, certainly, do not have the time or the ability.

So for the foreseeable future - GDAL is MONO only and that limits things a lot

**Comment by runette** [comment]
@develmusa I have created a HIGHLY EXPERIMENTAL version that in initial tests is working with IL2CPP.

If you are still interested - it would be good to get you to try it. A large part of this will be to test as much as possible - so I am sure that there will be some error messages I just don't know what they are yet.

To use this EXPERIMENTAL version - go to `Package Manager` and `Install from Git Url` and use the following Url

```
https://github.com/ViRGIS-Team/gdal-upm.git#change-to-swig-scripts
```

**Comment by develmusa** [comment]
that's very good news. thank you for the information. 

unfortunately, I don't have the time to test the version at the moment.

**Comment by runette** [comment]
From my tests, the experimental version is working with IL2CPP on the Windows and Mac players and on the UWP player.

So I have released the experimental version as release 3.4.102 and I am closing this issue. If there any problems with 3.4.102 or later in IL2CPP then please create new and specific issues.
